### ISSUE TITLE
[polaris.shopify.com] Address critical accessibility issues

### ISSUE THREAD

**Issue by alex-page** [Shopify/polaris-react-archive#6442]
We should be seen as leaders of performance, accessibility and much more. The current site is not in a state that demonstrates this.

- [ ] Heading order
- [ ] Hover styles
- [ ] Focus styles
- [ ] Keyboard navigation
- [x] Reduced motion
  - [x] #6469
- [ ] Color contrast
  - [x] #6470
  - [x] #6471
- [ ] via axe-core (axe DevTools): All page content should be contained by landmarks
  - [x] #6474
  - [ ] Update top nav markup to use a `<nav>` element in a `<header>` element
  - [ ] Update `<SideNav>` to be an `<aside>` and wrap the `<ul>` element in a `<nav>` element 
- [ ] Dynamic content should be marked up as live regions 
  - [ ] `ComponentPage`:
    - [ ] Set an `id, `aria-live="polite"`, and `role="region"` on the element wrapping the examples
    - [ ] Set `aria-controls` on the `select` to the `id` of the element wrapping the examples below it
  - [ ] `IconPage`: 
    - [x] #6508
    - [ ] Set an `id`, `aria-live="polite"`, and `role="region"` on the top element of the `SidebarContent`
    - [ ] Set `aria-controls` on the top element of the `IconGrid` to the `id` of the top element of the `SidebarContent`
  - [x] #6507
    - [ ] Set an `id, `aria-live="polite"`, and `role="region"` on the element wrapping the `IconGrid`
    - [ ] Set `aria-controls` on the search `input` to the `id` of the element wrapping the `IconGrid`

**Comment by ardakaracizmeli** [comment]
Hey there, this is also an issue for the color contrast of the icon names displayed on the icon names under:
[https://polaris.shopify.com/icons](https://polaris.shopify.com/icons)

**Comment by martenbjork** [comment]
Changed the title of the issue to assume positive intent.

**Comment by jbalsas** [comment]
Not sure if this one's tracked anywhere else, but usage snippets don't seem to have enough contrast in light mode and can be hard to read. Just tracked some pages to provide enough different examples, but I guess there are likely more occurrences than the ones listed below.

| Component | Dark Mode | Light Mode | 
| --- | --- | --- |
| [App Provider](https://polaris.shopify.com/components/app-provider) | <img width="811" alt="Screenshot 2022-07-14 at 10 38 06" src="https://user-images.githubusercontent.com/905006/178952734-6a5b9dc7-3176-47a1-9308-ee91f832188f.png"> | <img width="811" alt="Screenshot 2022-07-14 at 10 37 57" src="https://user-images.githubusercontent.com/905006/178952762-55b927bb-b4b7-4fdf-a0a3-60a3808e0a6d.png"> |
| [Button](https://polaris.shopify.com/components/button) | <img width="815" alt="Screenshot 2022-07-14 at 10 39 32" src="https://user-images.githubusercontent.com/905006/178953068-6e414efd-5022-4ec8-99ba-933fb3854c79.png"> | <img width="814" alt="Screenshot 2022-07-14 at 10 39 18" src="https://user-images.githubusercontent.com/905006/178953088-95a23a1d-620b-4bdb-9ead-908068fd1fa7.png"> |
| [Card](https://polaris.shopify.com/components/card) | <img width="824" alt="Screenshot 2022-07-14 at 10 31 28" src="https://user-images.githubusercontent.com/905006/178951621-40c19d3f-c68f-41b0-9294-fa2a6d82a677.png"> | <img width="822" alt="Screenshot 2022-07-14 at 10 31 18" src="https://user-images.githubusercontent.com/905006/178951640-177957bc-fbdf-49b7-8bb4-c73954a6ca0b.png"> |
| [Custom Properties](https://polaris.shopify.com/components/custom-properties) | <img width="813" alt="Screenshot 2022-07-14 at 10 33 48" src="https://user-images.githubusercontent.com/905006/178952093-54e824a0-c3b7-4605-bbb7-ea518e9c5da0.png"> | <img width="810" alt="Screenshot 2022-07-14 at 10 33 38" src="https://user-images.githubusercontent.com/905006/178952106-9b8be0f5-70e5-4fa2-a104-36e84aa8dcbc.png"> |
| [Keyboard Key](https://polaris.shopify.com/components/keyboard-key) | <img width="811" alt="Screenshot 2022-07-14 at 10 34 45" src="https://user-images.githubusercontent.com/905006/178952144-58820d64-af54-4fb5-bcfa-3743a4bd58d8.png"> | <img width="813" alt="Screenshot 2022-07-14 at 10 34 34" src="https://user-images.githubusercontent.com/905006/178952155-9b33e637-dbd3-47dd-8b19-3c1b6d1527e4.png"> |

**Comment by martenbjork** [comment]
Thanks @jbalsas! We accidentally broke these yesterday. A fix is up here: https://github.com/Shopify/polaris/pull/6630
